Makefile

```diff
@@ -477,44 +477,50 @@ $(LOCALBIN):
 
 ## Tools
 KUSTOMIZE ?= $(LOCALBIN)/kustomize
-# same as used in kubectl v1.28.x ; see https://github.com/kubernetes-sigs/kustomize/tree/master?tab=readme-ov-file#kubectl-integration
+
+# https://github.com/kubernetes/kubectl/blob/release-1.30/go.mod#L44
+# same as used in kubectl v1.30.x; https://github.com/kubernetes/kubectl/blob/release-1.30/go.mod#L44
 KUSTOMIZE_VERSION ?= v5.1.1
 
 ENVTEST ?= $(LOCALBIN)/setup-envtest
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 # List of available versions: https://storage.googleapis.com/kubebuilder-tools
-ENVTEST_K8S_VERSION = 1.28.3
+ENVTEST_K8S_VERSION = 1.30.0
 
 CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
 # We went ahead of k8s 1.28 with controller-tools v0.14.0 (which depends on k8s 1.29) to unblock go 1.22 upgrade.
 # It should be *relatively* safe as the only changes from this controller-tools version are description in CRD.
 # Once we upgrade to k8s 1.29, there's no need to change CONTROLLER_TOOLS_VERSION, and this 3 lines can be removed.
 #
 # k8s deps @ 1.29.0 https://github.com/kubernetes-sigs/controller-tools/blob/<version>/go.mod
-CONTROLLER_TOOLS_VERSION ?= v0.14.0
+CONTROLLER_TOOLS_VERSION ?= v0.15.0
 
-CODE_GENERATOR_VERSION ?= v0.28.12
+# Should match the kubeernetes minor vesion
+CODE_GENERATOR_VERSION ?= v0.30.7
 
 KUTTL ?= $(LOCALBIN)/kuttl
-# k8s deps @ 1.28.3
-KUTTL_VERSION ?= v0.16.0
+KUTTL_VERSION ?= v0.18.0
+
 ifeq ($(GOARCH), arm64)
     KUTTL_ARCH = arm64
 else ifeq ($(GOARCH), amd64)
     KUTTL_ARCH = x86_64
 else
     $(error Unsupported architecture: $(GOARCH))
 endif
+
 KUBECTL ?= $(LOCALBIN)/kubectl
-KUBECTL_VERSION ?= v1.29.10
+KUBECTL_VERSION ?= v1.30.7
+
+ETCD ?= $(LOCALBIN)/etcd
+ETCD_VERSION ?= v3.5.10
```

Comment:
For future upgrades — what are the requirements on the etcd version? how did you pick v3.5.10 ?

Reply:
Hate to admit there was no good reasoning for why this version. I think I just saw this release, tested it, and it was close to the latest so I left it alone. I couldn't find a compatibility matrix but now that I think about it, probably should use the version kube is tested against - so I'll update the version and comment